src/main/java/com/kobylynskyi/graphql/codegen/model/graphql/GraphQLResponseProjection.java

```diff
@@ -1,9 +1,35 @@
 package com.kobylynskyi.graphql.codegen.model.graphql;
 
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.StringJoiner;
+
 /**
  * The implementation class should basically contain the fields of the particular type which
  * should be returned back to the client.
  */
-public interface GraphQLResponseProjection {
+public abstract class GraphQLResponseProjection {
+
+    protected final Map<String, Object> fields = new LinkedHashMap<>();
+    protected final Map<String, GraphQLParametrizedInput> parametrizedInputs = new LinkedHashMap<>();
+
+    @Override
+    public String toString() {
```

Comment:
This was moved out from generated `ResponseProjection` classes because the code is identical for all of them.